jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/EnhancedMalformedURLException.java

```diff
@@ -0,0 +1,33 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util;
+
+import java.net.MalformedURLException;
+
+/**
+ * Enhanced {@link MalformedURLException} that allows for causes.
+ */
+public class EnhancedMalformedURLException extends MalformedURLException
+{
+    public EnhancedMalformedURLException(String msg)
+    {
+        super(msg);
+    }
+
+    public EnhancedMalformedURLException(String msg, Throwable cause)
+    {
+        super(msg);
+        initCause(cause);
```

Comment:
If we are really concerned about revealing the root cause, we could either:
a) use a system property to not initialise it here (yuck)
b) have options on our default ErrorHandler to reveal either:
     + No exception information
     + Exception type but not message or stack
     + Exception with stack, but no cause or suppressed
     + Exception with stack, causes and suppressions

I think b) would be a good (other PR) enhancement.